src/runtime/Object.cpp

```diff
@@ -2180,4 +2183,183 @@ Object::FastLookupSymbolResult Object::fastLookupForSymbol(ExecutionState& state
     return FastLookupSymbolResult();
 }
 
+bool DerivedObject::defineOwnProperty(ExecutionState& state, const ObjectPropertyName& P, const ObjectPropertyDescriptor& desc)
+{
+    if (UNLIKELY(isEverSetAsPrototypeObject() && !state.context()->vmInstance()->didSomePrototypeObjectDefineIndexedProperty() && P.isIndexString())) {
+        state.context()->vmInstance()->somePrototypeObjectDefineIndexedProperty(state);
+    }
+
+    // TODO Return true, if every field in Desc is absent.
```

Comment:
Can we call Object::defineOwnProperty(...) here?

Reply:
Good suggestion! I fixed it :)

Reply:
Oops, I made a mistake.
Since I added some check codes inside Object::defineOwnProperty to confirm that we check indexed property only for normal Objects, Object::defineOwnProperty could not be called at here.

Reply:
How about divide the codes into function and calling it?
IMHO too big code duplication for defineOwnProperty

Reply:
Thanks for the suggestion
I add another function (defineOwnPropertyMethod) to resolve large code duplication :)